vnext/Chakra/ChakraExecutor.cpp

```diff
@@ -299,6 +368,18 @@ void ChakraExecutor::initOnJSVMThread()
   // Add a pointer to ourselves so we can retrieve it later in our hooks
   JsSetContextData(m_context, this);
 
+#if !defined(USE_EDGEMODE_JSRT)
+  JsValueRef debuggerConsoleObject;
+  tls_runtimeTracker.DebugProtocolHandler->GetConsoleObject(&debuggerConsoleObject);
+
+  JsValueRef undefinedValue = JS_INVALID_REFERENCE;
+  JsGetUndefinedValue(&undefinedValue);
+
+  if (debuggerConsoleObject == JS_INVALID_REFERENCE || debuggerConsoleObject == undefinedValue) {
+    isPatchingRequired = false;
```

Comment:
I don't see where `isPatchingRequired` is declared. It should default to false, and we should set it to true when it is determined that the debugger should enabled debugger console redirection.

The name needs to be updated to better reflect the purpose: e.g. `needToRedirectConsoleToDebugger`. 

This should only be set to true if debugging is enabled. The code here to get the debugger console object should only happen if DebugProtocolHandler is not null. It appears this code was never tested without debugging being enabled because I think it would crash right now when this is null.

Reply:
Updated the code to set needToRedirectConsoleToDebugger to true, when both debugging and redirection are enabled.